src/float.cr

```diff
@@ -235,28 +235,28 @@ struct Float64
 
   def hash_normalize
     float_normalize_wrap do
-      {% if flag?(:x86) || flag?(:x86_64) || flag(:arm) || flag(:aarch64) %}
-	# it should work on every architecture where endianess of Float64 and Int64
-	# matches and float is IEEE754.
-	unsafe_int = unsafe_as(Int64)
-	exp = (((unsafe_int >> 52) & 0x7ff) - 1023)
-	mantisa = unsafe_int & ((1_u64 << 52) - 1)
-	if exp > -1023
-	  exp -= 52
-	  mantisa |= 1_u64 << 52
-	else
-	  # subnormals
-	  exp -= 51
-	end
-
-	{% if HASH_BITS == 31 %}
-	  mantisa %= HASH_MODULUS
-	{% end %}
-
-	{mantisa, exp}
-      {% else %}
-	float_normalize_reference
-      {% end %}
+      if FLOAT64_IS_IEEE754
```

Comment:
Sry, I meant if it can differ based on the system where the code is executed or is it fixed for every target architecture? If fixed, the appropriate branch should be chosen according to target arch at compile time.